src/snowflake/connector/connection.py

```diff
@@ -150,7 +150,7 @@ def DefaultConverterClass():
     "authenticator": (DEFAULT_AUTHENTICATOR, (type(None), str)),
     "mfa_callback": (None, (type(None), Callable)),
     "password_callback": (None, (type(None), Callable)),
-    "application": (CLIENT_NAME, (type(None), str)),
+    "application": (os.environ.get('SNOWFLAKE_PARTNER', CLIENT_NAME), (type(None), str)),
```

Comment:
We usually prefix env variables with `SF_`, let's keep to that.
```suggestion
    "application": (os.environ.get('SF_PARTNER', CLIENT_NAME), (type(None), str)),
```